doc/api/http2.md

```diff
@@ -2501,6 +2505,36 @@ req.on('end', () => {
 req.end('Jane');
 ```
 
+### The Extended CONNECT Protocol
+
+[RFC 8441][] defines an "Extended CONNECT Protocol" extension to HTTP/2 that
+may be used to bootstrap the use of an `Http2Stream` using the `CONNECT`
+method as a tunnel for other communication protocols (such as WebSockets).
+
+The use of the Extended CONNECT Protocol is enabled by HTTP/2 servers by using
+the `enableConnectProtocol` setting:
+
+```js
+const http2 = require('http2');
+const settings = { enableConnectProtocol: true };
+const server = http2.createServer({ settings });
+```
+
+Once the client receives the `SETTINGS` frame from the server indicating that
+the extended CONNECT may be used, it may send `CONNECT` requests that use the
+`':protocol'`  HTTP/2 pseudo-header:
+
+```js
+const http2 = require('http2');
+const client = http2.connect('http://localhost:8080');
+client.on('remoteSettings', (settings) => {
+  if (settings.enableConnectProtocol) {
+    const req = client.request({ ':method': 'CONNECT', ':protocol': 'foo' });
+    // ...
```

Comment:
Just when it got interesting ;) How does one use this feature? Is the idea that `req` is a raw Duplex stream now and another protocol can use it as its underlying transport stream?

Reply:
Effectively yeah... RFC 8441 says that the HTTP/2 Stream is to be used in place of the Socket... so here, the `Http2Stream` (on both client and server) is the underlying transport... layering FTW!

There are some challenges, however, in that the `ws` WebSocket implementation has a number of timing and API expectations built in that don't quite match up but, for the most part, it just kinda works.